libclc/generic/libspirv/async/async_work_group_strided_copy.inc

```diff
@@ -6,19 +6,17 @@
 //
 //===----------------------------------------------------------------------===//
 
-#define STRIDED_COPY(DST_AS, SRC_AS, DST_STRIDE, SRC_STRIDE)               \
-  size_t size = __spirv_LocalInvocationId_x() *                            \
```

Comment:
Do I understand correctly that hang happens because size can computed to 0 for some of the work items?

Reply:
yes, the update from ocl to spir-v inadvertently changed the step computation from "getting the workgroup size" to "mixing up all local ids". This bug was hidden so far as it relied on a header implementation (removed with https://github.com/intel/llvm/pull/1384).